### PR TITLE
fix(ui): use Optional for OIDC logout URL

### LINKED ISSUE #9594: OIDC logout URL config uses "f5" as magic default value instead of Optional
## Summary
Found something weird while reading through the UI config code. The `apicurio.ui.auth.oidc.logout-url` property has `"f5"` as its default value and then `SystemResourceImpl` checks against that string to decide whether to include `logoutUrl` in the response:

### `UserInterfaceConfigProperties.java`

```java
@ConfigProperty(name = "apicurio.ui.auth.oidc.logout-url", defaultValue = "f5")
public String authOidcLogoutUrl;
```

### `SystemResourceImpl.java`

```java
if (!"f5".equals(uiConfig.authOidcLogoutUrl)) {
    options.put("logoutUrl", uiConfig.authOidcLogoutUrl);
}
```

The other optional OIDC fields in the same class already use `Optional<String>` for exactly this (`authOidcRedirectUri`, `loadUserInfo`) this one just got missed. Should be a straightforward change to switch it over.

Introduced in #5199.

### PR DESCRIPTION
**Pull request by Kanishkaagrawal006**
Fixes #9594 
The OIDC logout URL configuration currently uses `"f5"` as a magic default value to represent an unset configuration, this change replaces that approach with `Optional<String>consistent.

### CODE REVIEW

File: app/src/main/java/io/apicurio/registry/rest/v3/impl/SystemResourceImpl.java [hunk 1]
```diff
@@ -104,9 +104,7 @@ private UserInterfaceConfigAuth uiAuthConfig() {
             uiConfig.authOidcRedirectUri.ifPresent(uri -> options.put("redirectUri", uri));
             options.put("clientId", uiConfig.authOidcClientId);
             options.put("scope", uiConfig.scope);
-            if (!"f5".equals(uiConfig.authOidcLogoutUrl)) {
-                options.put("logoutUrl", uiConfig.authOidcLogoutUrl);
-            }
+            uiConfig.authOidcLogoutUrl.ifPresent(logoutUrl -> options.put("logoutUrl", logoutUrl));
```
Comment (paoloantinori) [inline app/src/main/java/io/apicurio/registry/rest/v3/impl/SystemResourceImpl.java:107]: This wiring has no test pinning it. Please add or extend a test for the UI auth options that asserts both directions: `logoutUrl` is absent from the options map when `apicurio.ui.auth.oidc.logout-url` is unset, and present with the configured value when it is set. The `f5` default hid this contract for a long time, so an explicit assertion is what stops the next magic default from sneaking back in. I cannot see the test tree from this diff; if an existing `SystemResourceImpl` or UI config test exists, extending that is fine.

File: app/src/main/java/io/apicurio/registry/ui/UserInterfaceConfigProperties.java [hunk 2]
```diff
@@ -34,9 +34,9 @@ public class UserInterfaceConfigProperties {
     @ConfigProperty(name = "apicurio.ui.auth.oidc.client-id", defaultValue = "apicurio-registry-ui")
     @Info(category = CATEGORY_UI, description = "The OIDC clientId", availableSince = "3.0.0")
     public String authOidcClientId;
-    @ConfigProperty(name = "apicurio.ui.auth.oidc.logout-url", defaultValue = "f5")
+    @ConfigProperty(name = "apicurio.ui.auth.oidc.logout-url")
```
Comment (paoloantinori) [inline app/src/main/java/io/apicurio/registry/ui/UserInterfaceConfigProperties.java:37]: Removing the default changes behavior for any deployment that explicitly sets this property to `f5`: the old guard at the call site filtered that literal out, while the new `Optional` binding passes it through verbatim, so the UI would use `f5` as its logout URL. Before merging, grep the repo for other occurrences of the literal (helm charts, operator defaults, docker-compose files, test fixtures, docs) and check whether anyone documents `f5` as a way to disable logout. If the grep comes back clean, a short release-note line is all that is missing.

File: app/src/main/java/io/apicurio/registry/ui/UserInterfaceConfigProperties.java [hunk 3]
```diff
@@ -34,9 +34,9 @@ public class UserInterfaceConfigProperties {
...
     @Info(category = CATEGORY_UI, description = "The OIDC logout URL", availableSince = "3.0.0")
-    public String authOidcLogoutUrl;
+    public Optional<String> authOidcLogoutUrl;
```
Comment (paoloantinori) [inline app/src/main/java/io/apicurio/registry/ui/UserInterfaceConfigProperties.java:39]: One edge case worth a conscious decision and one more test assertion: the explicitly empty value. With the old `String` binding, setting this property to an empty string passed the `f5` check and emitted `logoutUrl` as an empty string into the UI options. With `Optional<String>` it may now resolve to an empty `Optional` (option dropped) or to an `Optional` holding the empty string, depending on the SmallRye version in use, and I cannot verify which from this diff. Dropping the option is the better outcome either way, so please confirm the actual behavior and cover it in the same test.

**Review by paoloantinori — COMMENTED** [review]
Direction is right: this removes a genuinely bad magic sentinel and makes `authOidcLogoutUrl` consistent with the sibling `Optional` fields (`authOidcRedirectUri`, `loadUserInfo`), and I traced the unset, set-to-real-URL, and set-to-`f5` cases to confirm the emitted options are unchanged for every sane configuration. Two points hold it up: the new unset-means-omitted contract has no test pinning it, and the `f5` sentinel was only filtered at this one call site, so anything outside these two files that pins the literal would now be emitted as a live logout URL.

- `app/src/main/java/io/apicurio/registry/rest/v3/impl/SystemResourceImpl.java:107`: the new `ifPresent` wiring is untested, so a regression back to a sentinel default would go unnoticed.
- `app/src/main/java/io/apicurio/registry/ui/UserInterfaceConfigProperties.java:37`: dropping `defaultValue = "f5"` silently changes meaning for deployments that explicitly set the property to `f5`; worth a repo-wide grep across charts, operator defaults, fixtures, and docs.
- `app/src/main/java/io/apicurio/registry/ui/UserInterfaceConfigProperties.java:39`: a public field changed type from `String` to `Optional<String>`; in-repo readers fail at compile time, but I could not check other modules or `@Info`-driven reflection from this diff alone.
- The explicitly-empty-value case (property set to blank) used to emit `logoutUrl` as an empty string; with `Optional<String>` it may now be dropped, which needs confirming plus one test assertion.

**Comment by Kanishkaagrawal006** [comment]
Added SystemResourceLogoutUrlUnsetTest and SystemResourceLogoutUrlSetTest  both passed locally, confirming logoutUrl is absent when unset and present with the configured value when set.

**Comment by Kanishkaagrawal006** [comment]
Grepped the repo for f5 and logout-url the only references outside these two files were in 3 auto-generated doc files -  assembly-config-reference.adoc, assembly-migrating-registry-v2-v3.adoc, ref-registry-all-configs.adoc  No Helm charts, docker-compose files, or fixtures reference it. Updated those 3 files to reflect optional<string> with no default, matching the existing redirect-uri entry pattern.

**Comment by Kanishkaagrawal006** [comment]
The other two OIDC settings in this same file - authOidcRedirectUri and loadUserInfo  are already Optional types, and their docs already generate correctly. So the doc-generation code clearly already knows how to handle Optional fields.

**Comment by Kanishkaagrawal006** [comment]
Added SystemResourceLogoutUrlEmptyTest to confirm the actual behavior: setting the property to an empty string resolves to Optional.empty(), so logoutUrl is correctly dropped from the options map

**Comment by paoloantinori** [comment]
Thanks — this closes out everything from my earlier review. The three new tests are exactly the right shape: they exercise the real getUIConfig() path and pin the contract behaviorally (absent when unset, absent when empty, present with the configured value when set), which covers both the untested-ifPresent point and the empty-string case I flagged. The doc regeneration matching how the sibling Optional fields generate is also the right call. No further comments from me.

**Comment by Kanishkaagrawal006** [comment]
Hi @paoloantinori 

Thank you  for the review!

There are two CI items that may need  maintainer attention:

License Compliance is failing with 4 issues. Could you point me toward what I should check?
Two workflows are “awaiting approval.” Could you or another maintainer approve them so they can run?

Please let me know if anything else is needed from my side. Thanks!

**Comment by paoloantinori** [comment]
On the two CI items:

1. License Compliance — nothing for you to fix here. I checked and the same "4 issues found" error is present on the main branch itself, so it is a pre-existing repo-wide condition, not something this PR introduced or can address.

2. Awaiting-approval workflows — I can see Verify and Validate Docs & Playbook sitting in action_required on this head. The catch is that the PR is currently in draft status, and the lifecycle orchestrator skips CI for drafts entirely, so approving those runs now would not achieve much. Mark it ready for review when you consider it final, and the workflows will re-enter the queue for a maintainer to approve.

**Comment by malladinagarjuna2** [comment]
looks good from my side too.
The old code used "f5" as a l default to mean "not configured" ,  an actual magic string a real deployment could collide with, and unreadable to anyone hitting it. [UserInterfaceConfigProperties] already has the right pattern one field up:
no issues all the  changes given by @paoloantinori  are addressed 
great work @Kanishkaagrawal006  
@carlesarnal @EricWittmann   good to merge

**Comment by malladinagarjuna2** [comment]
well the signoff by your own id is missiing 
thus prvalidation and validate pr is not passing.

**Comment by Kanishkaagrawal006** [comment]
Hii @paoloantinori
  I have fixed the DCO sign-off issue on the latest commits.Both commits now include the required Signed-off-by line and PR Validation and SonarCloud are passing.
The Verification Gate is currently blocked because lifecycle-ready=false, so the full verification suite has been skipped and Github is showing that an approving review is required.
Could you please approve the latest PR and re-trigger the lifecycle when you have a chance.
Thank you

**Review by paoloantinori — COMMENTED** [review]
Thanks for the follow-up. I re-checked all four of my earlier points and they hold up.

The `f5` literal appears nowhere else in the repo; the only other hits are sha256 substrings in `python-sdk/poetry.lock`. The operator routes this property through `putIfNotBlank` in `feat/security/Auth.java`, so it never emitted the sentinel in the first place. The empty-string behavior is confirmed rather than assumed: in smallrye-config, `STRING_CONVERTER` wraps `StringConverter` in `newEmptyValueConverter` with a null empty value, and `OptionalConverter` then wraps that null with `Optional.ofNullable`, so a blank value resolves to `Optional.empty()` and the option is dropped. That is what your Empty test asserts. The doc rows also match the generator's output for `Optional` fields, same as `apicurio.ui.auth.oidc.redirect-uri`.

Four things before this goes in.

The tests have never executed. The Verification Gate stops at `lifecycle-ready=false`, so Unit Tests is skipped and there is no evidence anywhere that the three new classes pass. Please run `./mvnw test -pl app -Dtest='SystemResourceLogoutUrl*Test'` locally and paste the output.

The profiles enable the OIDC tenant against `https://example.com/realms/test` with nothing behind it. Every other OIDC test here uses `AuthTestProfile` with the Keycloak testcontainer. Quarkus survives this, since the tenant falls back to a not-ready context after the connection timeout, but each test JVM still pays a real connection attempt against a public host. That is the kind of thing that flakes on a CI runner. Set `quarkus.oidc.discovery-enabled=false` with an explicit `jwks-path`, or drop OIDC from the cases that do not need it.

Three separate `@TestProfile` classes means three extra Quarkus restarts for one two-line contract, and two of them assert the identical thing. The unset and empty cases are really assertions about the `Optional<String>` binding rather than about `SystemResourceImpl`, so a single injection test on `UserInterfaceConfigProperties.authOidcLogoutUrl` covers both, leaving one `@QuarkusTest` for the set case.

While you are in there: add `assertEquals("test-client", options.get("clientId"))` next to the absence assertion so the test proves the OIDC branch actually ran, fix the PR description where it cuts off at "replaces that approach with `Optional<String>consistent.", drop the `@author kanu` tag from the Unset test, and add trailing newlines to the five files missing them.

The branch is 15 commits behind main, so it needs a rebase too. The milestone and the verification gate are on us rather than on you, and I will take care of those.
